packages/core/test/lib/console-child.js

```diff
@@ -0,0 +1,68 @@
+const assert = require("chai").assert;
+const fs = require("fs-extra");
+const path = require("path");
+const tmp = require("tmp");
+const TruffleConfig = require("@truffle/config");
+const { deriveConfigEnvironment } = require("../../lib/command-utils");
+
+let config;
+
+function createSandbox(source) {
+  if (!fs.existsSync(source)) {
+    throw new Error(`Sandbox failed: source: ${source} does not exist`);
+  }
+
+  const tempDir = tmp.dirSync({ unsafeCleanup: true });
+  fs.copySync(source, tempDir.name);
+  const config = TruffleConfig.load(
+    path.join(tempDir.name, "truffle-config.js"),
+    {}
+  );
+  return config;
+}
+
+describe("console-child", function () {
+  before("Create a sandbox", function () {
+    config = createSandbox(
+      path.join(__dirname, "..", "sources", "console-child")
+    );
+  });
+
+  describe("deriveConfigEnvironment", function () {
+    it("network has provider", function () {
+      config.network = "crazyTimeNetwork";
+      const cfg = deriveConfigEnvironment(config, config.network, "");
+      const expectedNetworkConfig = config.networks.crazyTimeNetwork;
+      assert.equal(cfg.network, config.network);
+      assert.equal(cfg.networks.crazyTimeNetwork, expectedNetworkConfig);
```

Comment:
Does this line satisfies your query @cds-amal ? If the `crazyTimeNetwork` has other properties, then it should be included in the `cfg.networks.crazyTimeNetwork`. Please correct me if I am wrong somewhere :)

Reply:
Hey @amal. Please ignore the above comment. I realised that there were some mistakes in the unit tests. I have updated the logic for each of the use cases.

Reply:
Hey @cds-amal. I realised that there were a few mistakes in the unit tests. I have updated the logic. Please ignore my above comment.